### DISCUSSION TITLE
PyPO not installable : no requirements for pip install

### DISCUSSION THREAD

**Discussion by dirkvictor** [PyPO-dev/PyPO#172] [Q&A] [answered]
On attempting to install PyPO according instructions in the manual 
- cmake 3.27.5 and wheel 0.37.1 installed OK
- listed dependencies for GUI installed OK
- cloned repository OK
Further info 
- using Ubuntu 22.04 , Python 3.10.6

Problem : the manual instructs to install by using 'pip install' without mentioning the requirements file. There is no requirements.txt or configure file in the cloned repository. I have also tried to install v1.0.1 (latest release) in the same way, but there is also no file with requirements.

Q : how to install PyPO after cloning the repository?

Dirk VICTOR
dirk.victor@chemipro.be

**Comment by arendMoerman** [comment, marked as answer]
Hi Dirk, 

After cloning the repository, PyPO can be installed simply by navigating to the cloned repository root directory and running:
```
pip install .
```
Just tested this in a new virtual environment (Debian 12), and runs without trouble.

There is no requirements.txt or setup.cfg file, because all dependencies are listed in the setup.py file.
Upon running above command, these are fetched and installed automatically, hence the lack of, and need for, specific requirements.txt and/or setup.cfg files.

Can you let us know here if this answers your question?

**Comment by dirkvictor** [comment]
I was fooled by the space and . after pipi install.

I performed the pip install command tincluding the pending space and . and it worked. Thanks.

Is it the purpose to always use jupyter notebook or can stand-alone python programs be run? Sorry , I am not a Linux geek but can work with python.



Dirk VICTOR
Chemipro

Leegstraat 76
9060 Zelzate

tel : (+32) 477 30 25 22


Op 23 september 2023 13:56:30 +02:00, schreef arend95 ***@***.***>:

**Comment by arendMoerman** [comment]
That is great to hear!

No, you can also run standalone programs. It is not necessary to put your scripts in notebooks.

**Comment by dirkvictor** [comment]
Again thanks for the quick and to-the-point support. I do appreciate.

For your info : I am introducing radio astronomy in an amateur astronomy club in Belgium. Detecting the hydrogen line at 1420 MHz with a home-built helical was not very satisfactory, so I am upgrading to a small dish. This stretches NEC2 to its maximum, and we want to install a larger dish. That is how I 'landed' on PyPO for further simulations.

Dirk VICTOR
Chemipro

Leegstraat 76
9060 Zelzate

tel : (+32) 477 30 25 22


Op 23 september 2023 15:06:26 +02:00, schreef arend95 ***@***.***>:

**Comment by arendMoerman** [comment]
No problem, and what a fun project you got going, it sounds really cool!
If you need any assistance writing your simulation, please don't hesitate to create a new discussion and we will gladly help you get on your way :)

Arend
